baseapp/grpcserver.go

```diff
@@ -22,6 +22,12 @@ import (
 
 // RegisterGRPCServer registers gRPC services directly with the gRPC server.
 func (app *BaseApp) RegisterGRPCServer(server gogogrpc.Server) {
+	app.RegisterGRPCServerWithSkipCheckHeader(server, false)
+}
+
+// RegisterGRPCServerWithSkipCheckHeader registers gRPC services with the specified gRPC server
+// and bypass check header flag.
+func (app *BaseApp) RegisterGRPCServerWithSkipCheckHeader(server gogogrpc.Server, skipCheckHeader bool) {
```

Comment:
can we add some text here about why this is useful?

i.e. gRPC queries may be processed during the commit phase, causing header checks to fail. we recommend not checking the header due to this. 

something like that

Reply:
also, come to think of it - is there a reason why anyone would want to have this enabled at all?